WireAPI/Tests/WireAPITests/APIs/SelfUserAPI/SelfUserAPITests.swift

```diff
@@ -0,0 +1,277 @@
+            // When
+            try await sut.pushSupportedProtocols([.mls])
+        }
+    }
+
+    func testPushSupportedProtocols_UnsupportedVersionError_V1() async throws {
+        // Given
+        let httpClient = HTTPClientMock(
+            code: 200,
+            payload: nil
+        )
+
+        let sut = SelfUserAPIV1(httpClient: httpClient)
+
+        // Then
+        await XCTAssertThrowsError(SelfUserAPIError.unsupportedEndpointForAPIVersion) {
+            // When
+            try await sut.pushSupportedProtocols([.mls])
+        }
+    }
+
+    func testPushSupportedProtocols_UnsupportedVersionError_V2() async throws {
+        // Given
+        let httpClient = HTTPClientMock(
+            code: 200,
+            payload: nil
+        )
+
+        let sut = SelfUserAPIV2(httpClient: httpClient)
+
+        // Then
+        await XCTAssertThrowsError(SelfUserAPIError.unsupportedEndpointForAPIVersion) {
+            // When
+            try await sut.pushSupportedProtocols([.mls])
+        }
+    }
+
+    func testPushSupportedProtocols_UnsupportedVersionError_V3() async throws {
+        // Given
+        let httpClient = HTTPClientMock(
+            code: 200,
+            payload: nil
+        )
+
+        let sut = SelfUserAPIV3(httpClient: httpClient)
+
+        // Then
+        await XCTAssertThrowsError(SelfUserAPIError.unsupportedEndpointForAPIVersion) {
+            // When
+            try await sut.pushSupportedProtocols([.mls])
+        }
+    }
+
+    func testPushSupportedProtocols_UnsupportedVersionError_V4() async throws {
+        // Given
+        let httpClient = HTTPClientMock(
+            code: 200,
+            payload: nil
+        )
+
+        let sut = SelfUserAPIV4(httpClient: httpClient)
+
+        // Then
+        await XCTAssertThrowsError(SelfUserAPIError.unsupportedEndpointForAPIVersion) {
+            // When
+            try await sut.pushSupportedProtocols([.mls])
+        }
+    }
+
+    // MARK: - Response handling
+
+    // MARK: - V0
+
+    func testGetSelfUser_SuccessResponse_200_V0() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(
+            code: 200,
+            payloadResourceName: "GetSelfUserSuccessResponseV0"
+        )
+
+        let sut = SelfUserAPIV0(httpClient: httpClient)
+
+        // When
+        let result = try await sut.getSelfUser()
+
+        // Then
+        XCTAssertEqual(
+            result,
+            Scaffolding.selfUserV0
+        )
+    }
+
+    func testGetSelfUser_FailureResponse() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(code: 404, errorLabel: "not-found")
+        let sut = SelfUserAPIV0(httpClient: httpClient)
+
+        // Then
+        await XCTAssertThrowsError {
+            // When
+            try await sut.getSelfUser()
+        }
+    }
+
+    // MARK: - V5
+
+    func testGetSelfUser_SuccessResponse_200_V5() async throws {
```

Comment:
This endpoint didn't change in v5, but it did in v4. I think you only need to rename the test, resource name, and use the v4 api.